tests/aws/services/kms/test_kms.py

```diff
@@ -1098,6 +1098,114 @@ def test_key_rotation_status(self, kms_key, aws_client):
         aws_client.kms.disable_key_rotation(KeyId=key_id)
         assert aws_client.kms.get_key_rotation_status(KeyId=key_id)["KeyRotationEnabled"] is False
 
+    @markers.aws.validated
+    def test_rotate_key_on_demand_modifies_key_material(self, kms_create_key, aws_client, snapshot):
+        key_id = kms_create_key(KeyUsage="ENCRYPT_DECRYPT", KeySpec="SYMMETRIC_DEFAULT")["KeyId"]
+        message = b"test message 123 !%$@ 1234567890"
+
+        ciphertext_before = aws_client.kms.encrypt(
+            KeyId=key_id,
+            Plaintext=base64.b64encode(message),
+            EncryptionAlgorithm="SYMMETRIC_DEFAULT",
+        )["CiphertextBlob"]
+
+        rotate_on_demand_response = aws_client.kms.rotate_key_on_demand(KeyId=key_id)
+        snapshot.match("rotate-on-demand-response", rotate_on_demand_response)
+
+        ciphertext_after = aws_client.kms.encrypt(
+            KeyId=key_id,
+            Plaintext=base64.b64encode(message),
+            EncryptionAlgorithm="SYMMETRIC_DEFAULT",
+        )["CiphertextBlob"]
+
+        assert ciphertext_before != ciphertext_after
+
+    @markers.aws.validated
+    def test_rotate_key_on_demand_with_symmetric_key_and_automatic_rotation_disabled_preserves_automatic_rotation_schedule_attributes(
+        self, kms_key, aws_client, snapshot
+    ):
+        key_id = kms_key["KeyId"]
+
+        rotation_status_response_before = aws_client.kms.get_key_rotation_status(KeyId=key_id)
+
+        rotate_on_demand_response = aws_client.kms.rotate_key_on_demand(KeyId=key_id)
+        snapshot.match("rotate-on-demand-response", rotate_on_demand_response)
+
+        rotation_status_response_after = aws_client.kms.get_key_rotation_status(KeyId=key_id)
+        assert (
+            rotation_status_response_after["KeyRotationEnabled"]
+            == rotation_status_response_before["KeyRotationEnabled"]
+        )
+        snapshot.match("rotation-status-response-after-rotation", rotation_status_response_after)
+
+    @markers.aws.validated
+    def test_rotate_key_on_demand_with_symmetric_key_and_automatic_rotation_enabled_preserves_automatic_rotation_schedule_attributes(
+        self, kms_key, aws_client, snapshot
+    ):
+        key_id = kms_key["KeyId"]
+
+        aws_client.kms.enable_key_rotation(KeyId=key_id)
+        rotation_status_response_before = aws_client.kms.get_key_rotation_status(KeyId=key_id)
```

Comment:
nit: same as above, we should snapshot this to avoid all the manual assert under